### ISSUE TITLE
TCP6ClientEndpoint doesn't implement interfaces.IStreamClientEndpoint

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5876]
|[<img alt="ashfall's avatar" src="https://avatars.githubusercontent.com/u/1138873?s=50" width="50" height="50">](https://github.com/ashfall)| @ashfall reported|
|-|-|
|Trac ID|trac#5876|
|Type|defect|
|Created|2012-08-17 16:59:03Z|
|Branch|https://github.com/twisted/twisted/tree/tcp6clientendpoint-implement-interface-5876|

The endpoint needs to implement the interface. Also, ClientEndpointTestCaseMixin should have a test that checks if the endpoints provide interfaces.IStreamClientEndpoint.

<details><summary>Searchable metadata</summary>

```
trac-id__5876 5876
type__defect defect
reporter__ashfall ashfall
priority__highest highest
milestone__ 
branch__branches_tcp6clientendpoint_implement_interface_5876 branches/tcp6clientendpoint-implement-interface-5876
branch_author__ashfall ashfall
status__closed closed
resolution__fixed fixed
component__core core
keywords__endpoint endpoint
time__1345222743000000 1345222743000000
changetime__1347189110000000 1347189110000000
version__None None
owner__ashfall ashfall
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="ashfall's avatar" src="https://avatars.githubusercontent.com/u/1138873?s=50" width="50" height="50">](https://github.com/ashfall)<a name="note_1"></a>|@ashfall commented|
|-|-|

(In [35266]) Branching to 'tcp6clientendpoint-implement-interface-5876'

**Comment by twisted-trac** [comment]
|[<img alt="ashfall's avatar" src="https://avatars.githubusercontent.com/u/1138873?s=50" width="50" height="50">](https://github.com/ashfall)<a name="note_3"></a>|@ashfall removed owner|
|-|-|

[Build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/tcp6clientendpoint-implement-interface-5876)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_4"></a>|@thijstriemstra set owner to @ashfall|
|-|-|

Thanks, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="ashfall's avatar" src="https://avatars.githubusercontent.com/u/1138873?s=50" width="50" height="50">](https://github.com/ashfall)<a name="note_5"></a>|@ashfall set status to `closed`|
|-|-|

(In [35559]) Merge tcp6clientendpoint-implement-interface-5876: TCP6ClientEndpoint now implements twisted.internet.interfaces.IStreamClientEndpoint

Author: ashfall
Reviewer: thijs
Fixes: [#5876](https://github.com/twisted/twisted/issues/5876)
